vcs/git.go

```diff
@@ -0,0 +1,42 @@
+package vcs
+
+import (
+	"gopkg.in/src-d/go-git.v4"
+)
+
+type GitRepository struct {
+	r   *git.Repository
+	dir string
+}
+
+// NewGitRepository takes the directory where a git repository exists and returns an implementation of the
+// System interface that uses the repository's metadata.
+func NewGitRepository(dir string) (*GitRepository, error) {
+	r, err := git.PlainOpen(dir)
+	if err == git.ErrRepositoryNotExists {
+		return nil, err
+	}
+	return &GitRepository{
+		r:   r,
+		dir: dir,
+	}, nil
+}
+
+func (gr *GitRepository) Head() Revision {
+	ref, err := gr.r.Head()
+	if err != nil {
+		panic(err)
```

Comment:
I definitely see now that `log.Fatal` is what the CLI tends to do rather than panic in similar circumstances. I'll revise this. But I feel that returning empty strings on failure is just failing _silently_. The CLI can't do its job if it doesn't have `Head`. What's it going to be doing with those empty string? Or am I missing something?

Reply:
`""` is the zero value for string and is acceptable to return here in my opinion. This method is only utilized by the function below which would ultimately return `""` as the revision.  I would `log.Warn` that a revision could not be found and return an empty string.
```
func Revision() string {
	inferred := ""
	if repo != nil {
		inferred = repo.Head().RevisionID
	}
	return TryStrings(StringFlag(flags.Revision), file.Revision(), inferred)
}
```

Reply:
Ok. Note that I kept the logic as it was before in `func (gr *GitRepository) Project() string`. Currently I think returning `""` from `Project()` is mostly fine, though I'd want to perhaps list [_all_ remotes](https://godoc.org/gopkg.in/src-d/go-git.v4#Repository.Remotes) for git and select one even if there isn't a remote named "origin". What do you think?

Reply:
also I'll move the logic to retrieve `HEAD` from git into the constructor `NewGitRepository` so we can simply return an error at that point if something goes wrong, and this will be consistent with how `NewSubversionRepository` will work.

Reply:
I think that would probably be the more correct solution, although I would be concerned about introducing a regression in this case. Its difficult to know if any customers are relying on our behavior when we don't find "origin".